### PR TITLE
General improvements to the cloud run frontend

### PR DESCRIPTION
**Pull request by another-rex**
A large list of small improvements to the frontend.

- Remove highlight.js dependency since it's only used to slightly highlight the schema on the home page, but takes up 75%+ of the javascript size. 
- Pre-compress all static files and serve the compressed version, reduces traffic by 75%+
- Add max age of 1 day (vs 10 mins before in appengine) to static files to reduce the number of loads. (new updates are still pushed out immediately because of the different file names each release)
- Compress dynamic content with Flask-Compress when the response is generated
- Remove font-ligatures (the things that make `affected` look weird because `ff` becomes one character)
- Make line-height larger and consistent with blog line height in vuln description
- Improve accessibility by adding various aria-labels, 
- move font loading to the bottom of the page to prevent blocking, 
- Add meta description
- Set height and width for images

### CODE REVIEW

**Comment by another-rex** [comment]
Original page speed analysis: https://pagespeed.web.dev/analysis/https-osv-dev/ojbx1vcw7t?form_factor=desktop
image

---

New page speed analysis: https://pagespeed.web.dev/analysis/https-osv-website-ivqje5d2hq-wl-a-run-app/c9f1tehq19?form_factor=desktop
image

File: gcp/appengine/blog/content/posts/almalinux-and-rocky-linux-join-osv.md [hunk 1]
```diff
@@ -10,4 +10,47 @@ Two new Linux distributions have been added to the OSV database. With the additi
 
 AlmaLinux and Rocky Linux were both started in response to CentOS moving upstream of Red Hat Enterprise Linux® (RHEL). These distros are open source, community driven, and 100% compatible with RHEL. Both maintainer teams worked closely with the OSV team to bring their advisories into the database. We’d like to thank them for their hard work! We know our community will benefit from their data contributions. 
 
-If you work with a project (like a Linux distribution) and would like to contribute data, we’d love to help get your data into OSV. Details of the process are available [here](https://github.com/google/osv.dev/blob/master/CONTRIBUTING.md#contributing-data).
\ No newline at end of file
+If you work with a project (like a Linux distribution) and would like to contribute data, we’d love to help get your data into OSV. Details of the process are available [here](https://github.com/google/osv.dev/blob/master/CONTRIBUTING.md#contributing-data).
+
```
Comment (andrewpollock) [inline gcp/appengine/blog/content/posts/almalinux-and-rocky-linux-join-osv.md:14]: This looks like an errant copy pasta?

**Review by andrewpollock — CHANGES_REQUESTED** [review]
Sick! Does any of this happen to fix the back button issues?

File: gcp/appengine/blog/content/posts/almalinux-and-rocky-linux-join-osv.md (hunk 1 above)
Comment (another-rex) [reply gcp/appengine/blog/content/posts/almalinux-and-rocky-linux-join-osv.md]: Ah yep, removed now.

**Comment by another-rex** [comment]
> Sick! Does any of this happen to fix the back button issues?

Sadly nope, I think that's related to turbo, might not be fixable without an upstream change.

File: gcp/appengine/Pipfile [hunk 2]
```diff
@@ -6,6 +6,7 @@ name = "pypi"
 [packages]
 Flask = "==2.3.2"
 Flask-Caching = "==2.0.2"
+Flask-Compress = "*"
```
Comment (michaelkedar) [inline gcp/appengine/Pipfile:9]: nit: pin these dependencies for consistency

File: gcp/appengine/Pipfile [hunk 3]
```diff
@@ -20,6 +21,7 @@ redis = "==4.6.0"
 requests = "==2.31.0"
 grpcio-status = "==1.56.2"
 gunicorn = "==21.2.0"
+whitenoise = "*"
```
Comment (michaelkedar) [inline gcp/appengine/Pipfile:24]: nit: pin these dependencies for consistency

File: gcp/appengine/frontend3/package.json [hunk 4]
```diff
@@ -33,6 +32,7 @@
     "sass-loader": "12.6.0",
     "style-loader": "3.3.3",
     "webpack": "5.87.0",
+    "webpack-bundle-analyzer": "^4.9.0",
```
Comment (michaelkedar) [inline gcp/appengine/frontend3/package.json:35]: nit: pin these dependencies for consistency

File: gcp/appengine/package-lock.json [hunk 5]
```diff
@@ -0,0 +1,6 @@
+{
```
Comment (michaelkedar) [inline gcp/appengine/package-lock.json:1]: Did you mean to create this file?

File: gcp/appengine/frontend3/src/templates/home.html [hunk 6]
```diff
@@ -63,7 +65,8 @@ <h2 class="heading">OSV schema</h2>
       </p>
       <div class="mdc-layout-grid__inner">
         <div class="mdc-layout-grid__cell--span-12">
-          <code-block>{
+          <!-- This is built with hugo -->
```
Comment (michaelkedar) [inline gcp/appengine/frontend3/src/templates/home.html:68]: Is this still being built by hugo?

**Review by michaelkedar — COMMENTED** [review]
Nice!
Will note that the changes to compression on `/static` will only occur on the Cloud Run website mirror, since App Engine bypasses the Flask app entirely for the files in that path.

File: gcp/appengine/package-lock.json (hunk 5 above)
Comment (another-rex) [reply gcp/appengine/package-lock.json]: Nope, removed now.

File: gcp/appengine/frontend3/src/templates/home.html (hunk 6 above)
Comment (another-rex) [reply gcp/appengine/frontend3/src/templates/home.html]: Nope, removed the comment

**Review by andrewpollock — APPROVED** [review]

**Review by michaelkedar — APPROVED** [review]

**Pull request merged**